### PR TITLE
(chore):Prepare for3.0

### PR DESCRIPTION
**Pull request by thomaszurkan-optimizely**
## Summary
- Update changelog and gradle build to point to new release 3.0.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,58 @@
 # Optimizely Android X SDK Changelog
 
+## 3.0.0
+February 11, 2019
+
+The 3.0 release improves event tracking and supports additional audience targeting functionality.
```
Comment (nchilada) [inline CHANGELOG.md]: Nit: spacing

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
+  * This will also allow you to attribute events to variations from other Optimizely projects in your account, even though those experiments don't appear in the same datafile.
+  * Note that for results segmentation in Optimizely results, the user attribute values from one event are automatically applied to all other events in the same session, as long as the events in question were actually received by our backend. This behavior was already in place and is not affected by the 3.0 release.
+* Support for all types of attribute values, not just strings.
+  * All values are passed through to notification listeners.
+  * Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
```
Comment (nchilada) [inline CHANGELOG.md]:
```suggestion
  * Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite float, double, integer, or long in the inclusive range [-2⁵³, 2⁵³]. 
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,58 @@
...
+* Support for all types of attribute values, not just strings.
+  * All values are passed through to notification listeners.
+  * Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³]. 
+  * Strings, booleans, and valid numbers are relevant for audience conditions.
+* Support for additional matchers in audience conditions:
+  * An `exists` matcher that passes if the user has a non-null value for the targeted user attribute and fails otherwise.
+  * A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+  * `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
```
Comment (nchilada) [inline CHANGELOG.md]:
```suggestion
  * `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite float, double, integer, or long in the inclusive range [-2⁵³, 2⁵³].
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,58 @@
...
+* Updated Pull Request template and commit message guidelines.
+* When given an invalid datafile, the Optimizely client object now instantiates into a no-op state instead of throwing a `ConfigParseException`. This matches the behavior of the other Optimizely SDKs.
+* Only remove old variation assignments from the user profile service when there are >= 100 persisted assignments.
+
+### Breaking Changes:
+* Java 7 is no longer supported.
+* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The `ActivateNotificationListener` and `TrackNotificationListener` interfaces now receive user attributes as `Map<String, ?>` instead of `Map<String, String>`.
+* The Localytics integration now relies on the user profile service to attribute events to the user's variations.
+* Drops support for the `getVariableBoolean`, `getVariableDouble`, `getVariableInteger`, and `getVariableString` APIs. Please migrate to the Feature Management APIs which are available in all 2.x releases.
```
Comment (nchilada) [inline CHANGELOG.md]: Updated. Thanks for your help with this!
```suggestion
* Drops support for the `getVariableBoolean`, `getVariableDouble`, `getVariableInteger`, and `getVariableString` APIs. Please migrate to the Feature Management APIs which were introduced in the 2.x releases.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,58 @@
...
+
+### Bug Fixes:
+* Experiments and features can no longer activate when a negatively targeted attribute has a missing, null, or malformed value.
+  * Audience conditions (except for the new `exists` matcher) no longer resolve to `false` when they fail to find an legitimate value for the targeted user attribute. The result remains `null` (unknown). Therefore, an audience that negates such a condition (using the `"not"` operator) can no longer resolve to `true` unless there is an unrelated branch in the condition tree that itself resolves to `true`.
+* The existence of a feature test in an experimentation group no longer causes A/B tests in the same group to activate the same feature.
```
Comment (nchilada) [inline CHANGELOG.md]: I think this entry only applied to JavaScript!

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,58 @@
...
+* Support for empty user IDs.
+* Sourceclear flagged jackson-databind 2.9.4 fixed in 2.9.8
```
Comment (nchilada) [inline CHANGELOG.md]: Okay, I guess we didn't need to mention the buffer reader change in Java or Android. Thanks for vetting that!

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,58 @@
...
+  * A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+  * `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
+  * The original (`exact`) matcher can now be used to target booleans and valid numbers, not just strings.
+* Support for A/B tests, feature tests, and feature rollouts whose audiences are combined using `"and"` and `"not"` operators, not just the `"or"` operator.
+* Datafile-version compatibility check: The SDK will remain uninitialized (i.e., will gracefully fail to activate experiments and features) if given a datafile version greater than 4.
+* Updated Pull Request template and commit message guidelines.
+* When given an invalid datafile, the Optimizely client object now instantiates into a no-op state instead of throwing a `ConfigParseException`. This matches the behavior of the other Optimizely SDKs.
+* Only remove old variation assignments from the user profile service when there are >= 100 persisted assignments.
```
Comment (nchilada) [inline CHANGELOG.md]: Oops, no need to mention the 100-assignment threshold - looks like that was already released in 2.1.4.

Comment (nchilada) [inline CHANGELOG.md]: Can you confirm that

> * Support for graceful shutdown in the default, async event dispatcher.

was only relevant for Java and not Android?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,58 @@
...
+### Bug Fixes
+* fix User Profile Service.  Don't trim the user profile service unless there are over 100 experiments in a users UPS. This will be configurable in the future.
+
+## 2.1.3
+December 6th, 2018
+
+### Bug Fixes
+The attributes map is now copied to ensure there is no concurrency issues.
+
 ## 3.0.0-RC2
```
Comment (nchilada) [inline CHANGELOG.md]: I think we can actually remove the RC content from the changelog. That's what we did in optimizely/javascript-sdk#217, and I would be fine with doing that retroactively in optimizely/java-sdk too.

File: CHANGELOG.md (hunk 7 above)
Comment (thomaszurkan-optimizely) [reply CHANGELOG.md]: yes

File: CHANGELOG.md (hunk 8 above)
Comment (nchilada) [reply CHANGELOG.md]: @thomaszurkan-optimizely how do you feel about removing the RC changes? They're redundant with the notes for the official 3.0 release.

File: CHANGELOG.md (hunk 7 above)
Comment (nchilada) [reply CHANGELOG.md]: Hurray! Thanks for confirming! 🎉

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,57 @@
...
+* The Localytics integration now relies on the user profile service to attribute events to the user's variations.
+* Drops support for the `getVariableBoolean`, `getVariableDouble`, `getVariableInteger`, and `getVariableString` APIs. Please migrate to the Feature Management APIs which were introduced in the 2.x releases.
+
+### Bug Fixes:
+* Experiments and features can no longer activate when a negatively targeted attribute has a missing, null, or malformed value.
+  * Audience conditions (except for the new `exists` matcher) no longer resolve to `false` when they fail to find an legitimate value for the targeted user attribute. The result remains `null` (unknown). Therefore, an audience that negates such a condition (using the `"not"` operator) can no longer resolve to `true` unless there is an unrelated branch in the condition tree that itself resolves to `true`.
+* Support for empty user IDs.
+* Sourceclear flagged jackson-databind 2.9.4 fixed in 2.9.8
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: nit. SourceClear.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,57 @@
 # Optimizely Android X SDK Changelog
 
+## 3.0.0
+February 13, 2019
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: nit. February 14 now.

File: CHANGELOG.md (hunk 8 above)
Comment (thomaszurkan-optimizely) [reply CHANGELOG.md]: I feel that the changelog should be just that.... a log of changes... so, I would want to include all releases in it.

Comment (nchilada) [reply CHANGELOG.md]: Okay, fine by me if it’s fine with @mikeng-optimizely and @aliabbasrizvi!

**Pull request merged**